### PR TITLE
Fix fading locations for unnecessary code.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/44816

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Looking for a good place to add a test for this so we don't regress this.

File: src/EditorFeatures/Core/Implementation/Diagnostics/DiagnosticsClassificationTaggerProvider.cs [hunk 1]
```diff
@@ -43,7 +44,7 @@ internal partial class DiagnosticsClassificationTaggerProvider : AbstractDiagnos
         protected override IEnumerable<Option2<bool>> Options => s_tagSourceOptions;
 
         [ImportingConstructor]
-        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
```
Comment (sharwell) [inline src/EditorFeatures/Core/Implementation/Diagnostics/DiagnosticsClassificationTaggerProvider.cs:47]: ❗ MEF parts cannot be directly constructed in test code. I can help set up the appropriate MEF container for the test(s) in question.

Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/Implementation/Diagnostics/DiagnosticsClassificationTaggerProvider.cs:47]: This currently follows the pattern used this class' siblings in tests. I'd rather get this in, then fix things up afterwards. Or, if you can push to my branch that works for me. Ideally with the same changes to DiagnosticSquiggleTaggerProvider. 
Thanks!

Comment (sharwell) [reply src/EditorFeatures/Core/Implementation/Diagnostics/DiagnosticsClassificationTaggerProvider.cs:47]: I'll push a change to the branch. I'm working through the many remaining cases over time so for now I'll just make the minimum change to preserve correct use of this one type.

Comment (sharwell) [reply src/EditorFeatures/Core/Implementation/Diagnostics/DiagnosticsClassificationTaggerProvider.cs]: Fixed

**Pull request merged**